src/kernels/ipywidgets-message-coordination/scriptUriConverter.node.ts

```diff
@@ -0,0 +1,109 @@
+import { EventEmitter, Event, Uri } from 'vscode';
+import { ILocalResourceUriConverter } from './types';
+import * as path from '../../platform/vscode-path/path';
+import { inject, injectable } from 'inversify';
+import { IFileSystem } from '../../platform/common/platform/types.node';
+import { IExtensionContext } from '../../platform/common/types';
+import { getOSType, OSType } from '../../platform/common/utils/platform';
+import { sha256 } from 'hash.js';
+import { createDeferred, Deferred } from '../../platform/common/utils/async';
+import { traceInfo, traceError } from '../../platform/logging';
+import { getComparisonKey } from '../../platform/vscode-path/resources';
+/* eslint-disable @typescript-eslint/no-var-requires, @typescript-eslint/no-require-imports */
+const sanitize = require('sanitize-filename');
+
+@injectable()
+export class ScriptUriConverter implements ILocalResourceUriConverter {
```

Comment:
I also had to separate out the URI converter part from the ipyWidgetScriptSource because it has node specific stuff in it too (reading/writing to local disk)